src/Aspire.Dashboard/Components/Dialogs/TextVisualizerDialog.razor.cs

```diff
@@ -14,44 +16,53 @@ public partial class TextVisualizerDialog : ComponentBase
     private const string XmlFormat = "xml";
     private const string JsonFormat = "json";
     private const string PlaintextFormat = "text";
+    private static readonly JsonSerializerOptions s_serializerOptions = new() { WriteIndented = true };
 
     private List<SelectViewModel<string>> _options = null!;
+    private readonly HashSet<string?> _enabledOptions = [];
     private string _selectedOption = null!;
-
     private string _formattedText = string.Empty;
 
     private readonly string _openSelectFormatButtonId = $"select-format-{Guid.NewGuid():N}";
     private bool _isSelectFormatPopupOpen;
 
-    private static readonly JsonSerializerOptions s_serializerOptions = new() { WriteIndented = true };
-
-    protected override void OnInitialized()
+    protected override void OnParametersSet()
     {
-        _options =
-        [
+        _enabledOptions.Clear();
+        _enabledOptions.Add(PlaintextFormat);
+
+        _options = [
             new SelectViewModel<string> { Id = PlaintextFormat, Name = Loc[nameof(Resources.Dialogs.TextVisualizerDialogPlaintextFormat)] },
             new SelectViewModel<string> { Id = JsonFormat, Name = Loc[nameof(Resources.Dialogs.TextVisualizerDialogJsonFormat)] },
-            new SelectViewModel<string> { Id = XmlFormat, Name = Loc[nameof(Resources.Dialogs.TextVisualizerDialogXmlFormat)] },
+            new SelectViewModel<string> { Id = XmlFormat, Name = Loc[nameof(Resources.Dialogs.TextVisualizerDialogXmlFormat)] }
         ];
-    }
 
-    protected override void OnParametersSet()
-    {
-        // We don't know what format the string is in, but we can guess
-        if (TryFormatXml())
+        if (TryFormatJson())
+        {
+            _selectedOption = JsonFormat;
+            _enabledOptions.Add(JsonFormat);
+        }
+        else if (TryFormatXml())
         {
             _selectedOption = XmlFormat;
-            return;
+            _enabledOptions.Add(XmlFormat);
+        }
+        else
+        {
+            _selectedOption = PlaintextFormat;
+            _formattedText = Content.Text;
         }
+    }
 
-        if (TryFormatJson())
+    private ICollection<ResourceLogLine> GetLines()
+    {
+        var lines = Regex.Split(_formattedText, "(\r\n|\n)", RegexOptions.Compiled).ToList();
+        if (lines.Count > 0 && lines[0].Length == 0)
         {
-            _selectedOption = JsonFormat;
-            return;
+            lines.RemoveAt(0);
         }
 
-        _selectedOption = PlaintextFormat;
-        _formattedText = Content.Text;
+        return lines.Select((line, index) => new ResourceLogLine(index, line, false)).ToList();
```

Comment:
aside -- in the JSON formatting (existing code below) it passes `CommentHandling = JsonCommentHandling.Skip` ... should that be .Allow? Are JSON comments lost when formatting JSON?

Reply:
No, it is not allowed to be `Allow` - this will throw an exception in modern versions of .NET Core. JSON comments are unsupported, the only reason `Skip` is chosen over `Disallow` was to allow formatting of otherwise valid JSON.

Reply:
If I'm inspecting a value in order to debug my app I might lose trust in the tool if I found it was not displaying all the content.

Reply:
I fixed it here: [`fd2dc69` (#5018)](https://github.com/dotnet/aspire/pull/5018/commits/fd2dc694a3d073593b539330cb5bbdd37b0afdb2)

`JsonDocument` doesn't support comments. But the lower-level reader and writer does.